airgun/entities/contenthost.py

```diff
@@ -55,27 +59,73 @@ def execute_package_action(self, entity_name, action_type, value):
+        :param customize_values: Dict with custom actions to run. Mandatory if customize is True
+
+        :return: Returns a dict containing job status details
+        """
+        if customize_values is None:
+            customize_values = {}
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.module_streams.search('name = {} and stream = {}'.format(module_name, stream_version))
+        action_type = dict(is_customize=customize, action=action_type)
+        view.module_streams.table.row(
+            name=module_name, stream=stream_version)['Actions'].fill(action_type)
+        if customize:
+            view = JobInvocationCreateView(view.browser)
+            view.fill(customize_values)
+            view.submit.click()
+        view = JobInvocationStatusView(view.browser)
+        view.wait_for_result()
+        return view.read()
+
     def search_package(self, entity_name, package_name):
         """Search for specific package installed in content host"""
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
         view.packages_installed.search(package_name)
         return view.packages_installed.table.read()
 
-    def install_errata(self, entity_name, errata_id):
+    def search_module_stream(self, entity_name, module_name, stream_version=None, status='All'):
+        """Search for specific package installed in content host"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        if stream_version is None:
+            view.module_streams.search(module_name, status)
+        else:
+            query = 'name = {} and stream = {}'.format(module_name, stream_version)
+            view.module_streams.search(query, status)
+        return view.module_streams.table.read()
+
+    def install_errata(self, entity_name, errata_id, install_via=None):
         """Install errata on a content host
 
         :param name: content host name to apply errata on
         :param errata_id: errata id or title, e.g. 'RHEA-2012:0055'
-
+        :param str install_via: via which mean to install errata. Available
+        options: "katello", "rex", "rex_customize"
         :return: Returns a dict containing task status details
         """
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
         view.errata.search(errata_id)
         view.errata.table.row(id=errata_id)[0].widget.fill(True)
-        view.errata.apply_selected.fill('Apply Selected')
-        view.dialog.confirm()
-        view = ContentHostTaskDetailsView(view.browser)
-        view.progressbar.wait_for_result()
+        if install_via == 'rex_customize':
+            view.errata.apply_selected.fill('via remote execution - customize first')
+        elif install_via == 'rex':
+            view.errata.apply_selected.fill('via remote execution')
+        else:
+            view.errata.apply_selected.fill('Apply Selected')
+            view.dialog.confirm()
```

Comment:
Just to make sure - this dialog appears only for installations run through katello-agent, right? Remote execution don't display this dialog?

Reply:
Looks like I will have to add another conditional statement.

Reply:
I thought 'via Katello agent' is default, but this version looks OK for me as well.